From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Sat, 29 Jul 2023 11:20:35 +0200
Subject: [PATCH] Do not log disconnections with null id

License: GPL-3.0-only (https://www.gnu.org/licenses/gpl-3.0.html)
Gale - https://galemc.org

This patch is based on the following patch:
"Add option to suppress null ID disconnections"
By: Kevin Raneri <kevin.raneri@gmail.com>
As part of: Pufferfish (https://github.com/pufferfish-gg/Pufferfish)
Licensed under: GPL-3.0-only (https://www.gnu.org/licenses/gpl-3.0.html)

* Pufferfish description *

This can help to hide annoying scanning bots from showing up in console.

diff --git a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
index 3bd65187084aa1ba38dec6fc4e270aeca5db2fd1..585cb5aafa4992e00f996c97f966d016ffaad3b7 100644
--- a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
@@ -134,6 +134,14 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
     // Paper end
     public void disconnect(final Component component) {
         try {
+            // Gale start - Pufferfish - do not log disconnections with null id
+            if (!org.galemc.gale.configuration.GaleGlobalConfiguration.get().logToConsole.nullIdDisconnections && this.authenticatedProfile != null && this.authenticatedProfile.id() == null) {
+                String reasonString = component.getString();
+                if ("Disconnected".equals(reasonString) || Component.translatable("multiplayer.disconnect.generic").getString().equals(reasonString)) {
+                    return;
+                }
+            }
+            // Gale end - Pufferfish - do not log disconnections with null id
             LOGGER.info("Disconnecting {}: {}", this.getUserName(), component.getString());
             this.connection.send(new ClientboundLoginDisconnectPacket(component));
             this.connection.disconnect(component);
